src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,429 @@
+extern "C"
+{
+#include "firmware/app/control/trajectory_planner.h"
+}
+
+#include <gtest/gtest.h>
+
+#include <iostream>
+#include <vector>
+
+#include "firmware/shared/math/polynomial_1d.h"
+#include "firmware/shared/math/polynomial_2d.h"
+#include "firmware/shared/math/vector_2d.h"
+#include "math.h"
+#include "trajectory_planner.h"
+
+class TrajectoryPlannerTest : public testing::Test
+{
+   protected:
+    virtual void SetUp() {}
+
+    virtual void TearDown() {}
+
+    static std::vector<double> getVelocityFromTrajectory(TrajectoryElement_t trajectory[],
+                                                         int num_elements,
+                                                         double inital_speed)
+    {
+        std::vector<double> velocity;
+
+        // Calculate the delta-position and the delta-time
+        velocity.push_back(inital_speed);
+
+        for (unsigned i = 1; i < num_elements - 1; i++)
+        {
+            const double dx = trajectory[i + 1].position.x - trajectory[i].position.x;
+            const double dy = trajectory[i + 1].position.y - trajectory[i].position.y;
+
+            const double dt = trajectory[i + 1].time - trajectory[i].time;
+
+            velocity.push_back(sqrt((pow(dx, 2) + pow(dy, 2))) / dt);
+        }
+        return velocity;
+    }
+
+    static std::vector<double> getAccelerationFromVelocity(
+        std::vector<double> velocity, TrajectoryElement_t trajectory[], int num_elements)
```

Comment:
Don't we have a struct that holds ` TrajectoryElement_t trajectory[], int num_elements` that we could use here? same goes for `getVelocityFromTrajectory`?

Reply:
Yeah I made that afterwords, I swap it in here